tests/model_test.py

```diff
@@ -232,6 +234,76 @@ def test_forward(
     torch.testing.assert_close(output2.logits[0][-1], batch_output_from_cached.logits[1][-1], rtol=rtol, atol=atol)
 
 
+@pytest.mark.parametrize("positional_embeddings", (None, "rope", "alibi"))
+@pytest.mark.parametrize("block_type", (BlockType.sequential,))
+@pytest.mark.parametrize("multi_query_attention", (True, False))
+def test_flash_attn(
+    train_config: TrainConfig,
+    tokenizer: Tokenizer,
+    positional_embeddings: Optional[str],
+    block_type: BlockType,
+    multi_query_attention: bool,
+):
+    torch.manual_seed(0)
+    torch.use_deterministic_algorithms(True)
+    device = torch.device("cuda")
+
+    if positional_embeddings is None:
+        alibi = False
+        rope = False
+    elif positional_embeddings == "alibi":
+        alibi = True
+        rope = False
+    elif positional_embeddings == "rope":
+        alibi = False
+        rope = True
+    else:
+        raise ValueError(f"{positional_embeddings} is not a valid value for positional_embeddings")
+
+    def make_model(flash_attn: bool):
+        train_config.model.alibi = alibi
+        train_config.model.rope = rope
+        train_config.model.flash_attention = flash_attn
+        train_config.model.attention_dropout = 0.0
+        train_config.model.block_type = block_type
+        train_config.model.multi_query_attention = multi_query_attention
+        train_config.model.init_device = "cuda"
+        return Olmo(train_config.model).eval()
+
+    input1 = tokenizer.encode("As a large language model, I don’t have personal opinions.")
+    input2 = tokenizer.encode("But I can share some interesting facts!")
+    batch_inputs = DataCollator.from_train_config(train_config)(
+        [  # type: ignore
+            {"input_ids": input1, "attention_mask": [1.0] * len(input1)},
+            {"input_ids": input2, "attention_mask": [1.0] * len(input2)},
```

Comment:
I don't think we're actually using `flash-attn` here since we're passing an attention mask.

Reply:
I would just remove the mask here.

Reply:
Haha, yes, I noticed all those same issues and fixed them before I read your comment here 😮‍💨.